envsec/pkg/envcli/flags.go

```diff
@@ -82,9 +82,9 @@ func (f *configFlags) validateProjectID(orgID id.OrgID) (string, error) {
 }
 
 type CmdConfig struct {
-	Store    envsec.Store
-	EnvID    envsec.EnvID
-	EnvNames []string
+	envsec   *envsec.Envsec
```

Comment:
Returning the store felt like it was getting convoluted. It also required each caller to set the store individually. Long term this will need rethinking if we want to support more open source non-jetpack stores.

Longer term, I kinda imagine envsec working something like:

`envsec.New(ctx, store, envsec.WithAuthProvider(myAuthProvider))`

Which would allow you to mix and match stores and auth providers. Maybe stores have default auth providers as well. In that world, `genConfig` really just becomes a function to create a new envsec instance.